packages/stream-model-handler/src/interfaces-utils.ts

```diff
@@ -687,3 +690,16 @@ export async function validateImplementedInterfaces(
     throw new AggregateError(errors, `Interfaces validation failed for model ${model.name}`)
   }
 }
+
+export function isValidImmutabilityImplementation(
+  expected: ModelDefinition,
+  implemented: ModelDefinition
+): boolean {
+  const expectedModelV2 = expected as ModelDefinitionV2
```

Comment:
Sorry if my previous comment wasn't clear but I meant actually having a runtime check for the definition version, ex `if (expected.version === '2.0')`, not just replacing the type here.